generators/new_collateral/config/parser_test.go

```diff
@@ -0,0 +1,140 @@
+package config_test
+
+import (
+	"github.com/makerdao/vdb-mcd-transformers/generators/new_collateral/config"
+	"github.com/makerdao/vdb-mcd-transformers/generators/new_collateral/test_data"
+	"github.com/makerdao/vdb-mcd-transformers/generators/new_collateral/types"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Config Parser", func() {
+	var (
+		testConfigFilePath   = "../test_data/"
+		testConfigFileName   = "testConfig"
+		badTestConfigFile    = "non-existent-file"
+		configParser         = config.NewParser()
+		expectedParsedConfig = types.TransformersConfig{
+			ExporterMetadata: types.ExporterMetaData{
+				Home:             "github.com/makerdao/vulcanizedb",
+				Name:             "transformerExporter",
+				Save:             false,
+				Schema:           "maker",
+				TransformerNames: []string{"cat_v1_1_0", "cat_file_vow"},
+			},
+			TransformerExporters: types.TransformerExporters{
+				"cat_v1_1_0":   test_data.Cat110Exporter,
+				"cat_file_vow": test_data.CatFileVowExporter,
+			},
+			Contracts: types.Contracts{
+				"MCD_CAT_1_0_0": test_data.Cat100Contract,
+				"MCD_CAT_1_1_0": test_data.Cat110Contract,
+			},
+		}
+	)
+
+	Context("ParseCurrentConfigFile", func() {
+		It("reads in the existing config file", func() {
+			config, parseErr := configParser.ParseCurrentConfig(testConfigFilePath, testConfigFileName)
+			Expect(parseErr).NotTo(HaveOccurred())
+			Expect(config).To(Equal(expectedParsedConfig))
+		})
+
+		It("returns an error if it fails to decode the file", func() {
+			_, parseErr := configParser.ParseCurrentConfig(testConfigFilePath, badTestConfigFile)
+			Expect(parseErr).To(HaveOccurred())
+			Expect(parseErr).To(MatchError("open ../test_data/non-existent-file.toml: no such file or directory"))
+		})
+	})
+
+	Context("ParseExporterMetadata", func() {
```

Comment:
nbd and probably unnecessary, but is there a reason we can't make these private and test them through `ParseCurrentConfigFile`?

Reply:
👍  planning to take a look at this when working on getting a new PR together